File: POOEncapsulamento/app/src/main/java/br/com/app/pooencapsulamento/MainActivity.java
package br.com.app.pooencapsulamento;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private TextView txtTituloLivro;
    private TextView txtAutor;
    private TextView txtEditora;
    private TextView txtPaginas;
    private TextView txtLocalizacao;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        iniciarComponentesDoLayout();

        criarObjetos();
    }

    private void iniciarComponentesDoLayout() {
        txtAutor = findViewById(R.id.txtAutor);
        txtTituloLivro = findViewById(R.id.txtTituloLivro);
        txtEditora = findViewById(R.id.txtEditora);
        txtPaginas = findViewById(R.id.txtPaginas);
        txtLocalizacao = findViewById(R.id.txtLocalizacao);
    }

    public void criarObjetos() {
        Livro livro = new Livro()
        livro.setTitulo("O poder da paciência");
        livro.setAutor("Mary Jane Ryan");
        livro.setEditora("L.S Lakers");
        livro.setPaginas(110);
        livro.setLocalizacao("Nova York");

        Log.i("Livro ", "Obj Livro: Titulo, "+ livro.getTitulo());
        Log.i("Livro ", "Obj Livro: Autor, "+ livro.getAutor());
        Log.i("Livro ", "Obj Livro: Editora, "+ livro.getEditora());
        Log.i("Livro ", "Obj Livro: Paginas, "+ livro.getPaginas());
        Log.i("Livro ", "Obj Livro: Localização, "+ livro.getLocalizacao());

        txtAutor.setText(livro.getAutor());
        txtTituloLivro.setText(livro.getTitulo());
        txtEditora.setText(livro.getEditora());
        txtPaginas.setText(livro.getPaginas()+"");
        txtLocalizacao.setText(livro.getLocalizacao());
    }
}

File: POOKotlinEncapsulamento/app/src/main/java/br/com/app/pookotlinencapsulamento/MainActivity.kt
package br.com.app.pookotlinencapsulamento

import android.os.Bundle
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    lateinit var txtTitulo: TextView
    lateinit var txtAutor: TextView
    lateinit var txtLocalizacao: TextView
    lateinit var txtEditora: TextView
    lateinit var txtPaginas: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        iniciarComponentesDoLayout()

        criarObjetos()
    }

    private fun iniciarComponentesDoLayout() {
        txtTitulo = findViewById(R.id.txtTituloLivro)
        txtAutor = findViewById(R.id.txtAutor)
        txtLocalizacao = findViewById(R.id.txtLocalizacao)
        txtEditora = findViewById(R.id.txtEditora)
        txtPaginas = findViewById(R.id.txtPaginas)
    }

    private fun criarObjetos() {
        var livro = Livro("O poder da paciência", "Mary Jane Ryan", "Nova York",
            "L.S Lakers", 110)

        txtTitulo.setText(livro.titulo)
        txtAutor.setText(livro.autor)
        txtLocalizacao.setText(livro.localizacao)
        txtEditora.setText(livro.editora)
        txtPaginas.setText(livro.paginas.toString())
    }
}